lib/Google/RestApi.pm

```diff
@@ -32,32 +34,51 @@ sub new {
   my $class = shift;
 
   state $check = compile_named(
-    config_file => Str, { optional => 1 },
-    _extra_     => slurpy Any,
+    config_file          => Str, { optional => 1 },
+    service_account_file => Str, { optional => 1 },
+    _extra_              => slurpy Any,
   );
   my $self = named_extra($check->(@_));
 
+
   if ($self->{config_file}) {
     my $config = eval { LoadFile($self->{config_file}) };
     die "Unable to load config file '$self->{config_file}': $@" if $@;
     $self = Hash::Merge::merge($self, $config);
+  } elsif ($self->{service_account_file}) {
+    die "Unable to find Service Account JSON file at: $self->{service_account_file}" unless -e $self->{service_account_file};
   }
 
+
+
   state $check2 = compile_named(
-    config_file   => Str, { optional => 1 },
-    client_id     => Str,
-    client_secret => Str,
-    token_file    => Str,
-    timeout       => Int, { default => 120 },
-    throttle      => Int->where('$_ > -1'), { default => 0 },
-    post_process  => CodeRef, { optional => 1 },
+    config_file          => Str, { optional => 1 },
```

Comment:
I should have allowed for more than one auth type for this, and your pull request is exposing this weakness. I think config_file should be renamed to oauth2_client_file so it's more obvious to the user what's expected in the config file. Also, client_id, client_secret, and token_file should be under 'oauth2_client' hash key. so then you either supply an oauth2_client hash, or an oauth2_client_file, or a service_account_file, all of which is much more descriptive. Is this something you want to incorporate in this pr?

Reply:
Actually I would have to recant that... the config_file was intended to hold any and all possible keys, not just oauth. I'll give this some more thought.